### DISCUSSION TITLE
AI Verify Playground Error

### DISCUSSION THREAD

**Discussion by tempodemobob** [aiverify-foundation/aiverify#290] [Q&A]
I encountered this error using the custom report (report only robustness toolbox results) and sample image files provided by aiverify.
Any idea how to resolve the error?
Logs below:

2024-05-30 21:52:23,403 [INFO][app_logger.py::add_to_log(116)] [task_processing.py:run_task_processing_in_process(66)]: The task validation is successful: task:665882e197f3bef638a9f620-6658849797f3bef638b149ae
2024-05-30 21:52:23,403 [INFO][app_logger.py::add_to_log(116)] [task_processing.py:run_task_processing_in_process(71)]: Working on task: message_id 1717077143344-0, message_args {"mode":"upload","id":"task:665882e197f3bef638a9f620-6658849797f3bef638b149ae","algorithmId":"algo:aiverify.stock.robustness_toolbox:robustness_toolbox","algorithmArgs":{"annotated_ground_truth_path":"/app/aiverify/uploads/data/pickle_pandas_fashion_mnist_annotated_labels_10.sav","file_name_label":"file_name"},"testDataset":"/app/aiverify/uploads/data/raw_fashion_image_10","modelFile":"/app/aiverify/uploads/model/multiclass_classification_image_mnist_fashion","groundTruthDataset":"/app/aiverify/uploads/data/pickle_pandas_fashion_mnist_annotated_labels_10.sav","modelType":"classification","groundTruth":"labels"}, task_type: TaskType.NEW
2024-05-30 21:52:23,408 [INFO][app_logger.py::add_to_log(116)] [task_result.py:set_status(146)]: The current task has changed status to TaskStatus.RUNNING
2024-05-30 21:52:23,414 [INFO][app_logger.py::add_to_log(116)] [task_processing.py:process_new_task(110)]: Sending task update
2024-05-30 21:52:23,429 [INFO][app_logger.py::add_to_log(116)] [stream_processing.py:detect_pipeline(30)]: Attempting to detect pipeline model from the given path
2024-05-30 21:52:23,429 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(55)]: Attempting to read pipeline: /app/aiverify/uploads/model/multiclass_classification_image_mnist_fashion
2024-05-30 21:52:23,434 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(83)]: Pipeline validation successful
2024-05-30 21:52:23,444 [INFO][app_logger.py::add_to_log(116)] [task.py:_send_task_update(278)]: Task has received notification to send task update
2024-05-30 21:52:23,449 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(93)]: Found these non python files: {'fashion_mnist_lr_pipeline.sav': '/app/aiverify/uploads/model/multiclass_classification_image_mnist_fashion/fashion_mnist_lr_pipeline.sav'}
2024-05-30 21:52:23,449 [INFO][app_logger.py::add_to_log(116)] [worker.py:_send_update(614)]: The update sent successfully: task:665882e197f3bef638a9f620-6658849797f3bef638b149ae:{'type': 'TaskResponse', 'status': 'Running', 'elapsedTime': 0, 'startTime': '2024-05-30T21:52:23.402775', 'output': '""', 'logFile': '/app/aiverify/test-engine-app/logs/task:665882e197f3bef638a9f620-6658849797f3bef638b149ae.log', 'taskProgress': 0}
2024-05-30 21:52:23,454 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(111)]: Attempting to deserialize pipeline: /app/aiverify/uploads/model/multiclass_classification_image_mnist_fashion
2024-05-30 21:52:23,606 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(138)]: Attempting to identify pipeline format: {class 'sklearn.pipeline.Pipeline'}
2024-05-30 21:52:23,611 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(149)]: Supported pipeline format: {class 'sklearn.pipeline.Pipeline'}, PipelinePluginType.SKLEARN[sklearn.pipeline.Pipeline]
2024-05-30 21:52:23,616 [INFO][app_logger.py::add_to_log(116)] [task_processing.py:_load_instances(269)]: Found the pipeline model. Loading pipeline instances
2024-05-30 21:52:23,621 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(56)]: Attempting to read data: /app/aiverify/uploads/data/raw_fashion_image_10
2024-05-30 21:52:23,626 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(81)]: Data validation successful
2024-05-30 21:52:23,632 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/5.png
2024-05-30 21:52:23,644 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/1.png
2024-05-30 21:52:23,654 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/9.png
2024-05-30 21:52:23,677 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/7.png
2024-05-30 21:52:23,689 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/3.png
2024-05-30 21:52:23,700 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/6.png
2024-05-30 21:52:23,710 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/2.png
2024-05-30 21:52:23,720 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/0.png
2024-05-30 21:52:23,730 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/8.png
2024-05-30 21:52:23,739 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/raw_fashion_image_10/4.png
2024-05-30 21:52:23,750 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(157)]: Consolidation results: {pandasdata.Plugin object at 0x7f05880a2090} {class 'imageserializer.Plugin'}
2024-05-30 21:52:23,755 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(181)]: Supported data format: DataPluginType.PANDAS[SerializerPluginType.IMAGE]
2024-05-30 21:52:23,759 [INFO][app_logger.py::add_to_log(116)] [stream_processing.py:load_data(75)]: Data Instance: {pandasdata.Plugin object at 0x7f05880a2090}, Data Deserializer: SerializerPluginType.IMAGE
2024-05-30 21:52:23,771 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(55)]: Attempting to read pipeline: /app/aiverify/uploads/model/multiclass_classification_image_mnist_fashion
2024-05-30 21:52:23,776 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(83)]: Pipeline validation successful
2024-05-30 21:52:23,789 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(93)]: Found these non python files: {'fashion_mnist_lr_pipeline.sav': '/app/aiverify/uploads/model/multiclass_classification_image_mnist_fashion/fashion_mnist_lr_pipeline.sav'}
2024-05-30 21:52:23,794 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(111)]: Attempting to deserialize pipeline: /app/aiverify/uploads/model/multiclass_classification_image_mnist_fashion
2024-05-30 21:52:23,804 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(138)]: Attempting to identify pipeline format: {class 'sklearn.pipeline.Pipeline'}
2024-05-30 21:52:23,809 [INFO][log_utils.py::log_message(31)] [pipeline_manager.py:read_pipeline_path(149)]: Supported pipeline format: {class 'sklearn.pipeline.Pipeline'}, PipelinePluginType.SKLEARN[sklearn.pipeline.Pipeline]
2024-05-30 21:52:23,814 [INFO][app_logger.py::add_to_log(116)] [stream_processing.py:load_pipeline(132)]: Pipeline Instance: {sklearnpipeline.Plugin object at 0x7f058805c450}, Pipeline Deserializer: SerializerPluginType.PICKLE
2024-05-30 21:52:24,501 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(56)]: Attempting to read data: /app/aiverify/uploads/data/pickle_pandas_fashion_mnist_annotated_labels_10.sav
2024-05-30 21:52:24,506 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(81)]: Data validation successful
2024-05-30 21:52:24,510 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(89)]: Attempting to deserialize data: /app/aiverify/uploads/data/pickle_pandas_fashion_mnist_annotated_labels_10.sav
2024-05-30 21:52:24,518 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(157)]: Consolidation results: {pandasdata.Plugin object at 0x7f058858f650} {class 'pickleserializer.Plugin'}
2024-05-30 21:52:24,523 [INFO][log_utils.py::log_message(31)] [data_manager.py:read_data(181)]: Supported data format: DataPluginType.PANDAS[SerializerPluginType.PICKLE]
2024-05-30 21:52:24,533 [INFO][app_logger.py::add_to_log(116)] [stream_processing.py:load_ground_truth(259)]: GroundTruth Instance: {pandasdata.Plugin object at 0x7f058858f650}, GroundTruth Deserializer: SerializerPluginType.PICKLE
2024-05-30 21:52:24,540 [ERROR][log_utils.py::log_message(37)] [algorithm_manager.py:get_algorithm(126)]: There was an error getting algorithm instance (not found): algo:aiverify.stock.robustness_toolbox:robustness_toolbox
2024-05-30 21:52:24,545 [INFO][app_logger.py::add_to_log(116)] [plugin_controller.py:get_plugin_instance(112)]: Attempting to find algo:aiverify.stock.robustness_toolbox:robustness_toolbox in the algorithm registry
2024-05-30 21:52:24,552 [INFO][app_logger.py::add_to_log(116)] [plugin_controller.py:get_plugin_instance(141)]: algo:aiverify.stock.robustness_toolbox:robustness_toolbox is in the algorithm registry. Attempting to re-discover algorithm
2024-05-30 21:52:24,580 [INFO][robustness_toolbox.py::add_to_log(205)] Setup completed
2024-05-30 21:52:24,589 [INFO][log_utils.py::log_message(31)] [algorithm_manager.py:get_algorithm(116)]: Supported algorithm: algo:aiverify.stock.robustness_toolbox:robustness_toolbox, PluginType.ALGORITHM
2024-05-30 21:52:24,593 [INFO][app_logger.py::add_to_log(116)] [stream_processing.py:load_algorithm(356)]: Algorithm Instance: {robustness_toolbox.Plugin object at 0x7f0575d309d0}
2024-05-30 21:52:24,610 [INFO][app_logger.py::add_to_log(116)] [task_result.py:set_progress(114)]: The current task completion progress is 100
2024-05-30 21:52:24,623 [INFO][app_logger.py::add_to_log(116)] [task_processing.py:process_new_task(147)]: The raw task results: {'results': [0]}
2024-05-30 21:52:24,626 [ERROR][app_logger.py::add_to_log(126)] [task_processing.py:process_new_task(167)]: Failed output schema validation: Task Results: {'results': [0]}
2024-05-30 21:52:24,629 [WARNING][app_logger.py::add_to_log(121)] [task_processing.py:process_new_task(189)]: The task terminated: The algorithm output schema validation failed
2024-05-30 21:52:24,638 [INFO][app_logger.py::add_to_log(116)] [task.py:_send_task_update(278)]: Task has received notification to send task update
2024-05-30 21:52:24,643 [INFO][app_logger.py::add_to_log(116)] [worker.py:_send_update(614)]: The update sent successfully: task:665882e197f3bef638a9f620-6658849797f3bef638b149ae:{'type': 'TaskResponse', 'status': 'Running', 'elapsedTime': 1, 'startTime': '2024-05-30T21:52:23.402775', 'output': '""', 'logFile': '/app/aiverify/test-engine-app/logs/task:665882e197f3bef638a9f620-6658849797f3bef638b149ae.log', 'taskProgress': 100}
2024-05-30 21:52:24,647 [INFO][app_logger.py::add_to_log(116)] [task_result.py:set_progress(114)]: The current task completion progress is 100
2024-05-30 21:52:24,648 [INFO][app_logger.py::add_to_log(116)] [task_result.py:set_status(146)]: The current task has changed status to TaskStatus.ERROR
2024-05-30 21:52:24,692 [INFO][app_logger.py::add_to_log(116)] [worker.py:_send_update(614)]: The update sent successfully: task:665882e197f3bef638a9f620-6658849797f3bef638b149ae:{'type': 'TaskResponse', 'status': 'Error', 'elapsedTime': 1, 'startTime': '2024-05-30T21:52:23.402775', 'output': '', 'errorMessages': '[{"category": "SYSTEM_ERROR", "code": "CSYSx00146", "description": "Task Terminated: The algorithm output schema validation failed", "severity": "warning", "component": "task_processing.py"}]', 'logFile': '/app/aiverify/test-engine-app/logs/task:665882e197f3bef638a9f620-6658849797f3bef638b149ae.log', 'taskProgress': 100}
2024-05-30 21:52:24,692 [INFO][app_logger.py::add_to_log(116)] [worker.py:_send_acknowledgement(659)]: The acknowledgement sent successfully - 1717077143344-0
2024-05-30 21:52:24,698 [INFO][app_logger.py::add_to_log(116)] [task.py:cleanup(104)]: The system has received notification to clean up task
